src/Application/Link.php

```diff
@@ -55,10 +59,14 @@ public static function fromMultilingualLink(
      */
     public static function fromArray(array $data): self
     {
+        /** @var Link|null $parentData */
+        $parentData = $data[MultilingualLink::PARENT] ?? null;
+
         return new self(
             $data[MultilingualLink::NAME],
             $data[MultilingualLink::LINK_TEXT],
             $data[MultilingualLink::LINK_URL],
+            !empty($parentData) ? self::fromArray($parentData) : null,
```

Comment:
I think this is wrong. If `$parentData` is not empty, then `fromArray` is called again but this time you 're passing a `Link` object instead of an array. Apart from the type error this (calling `fromArray()` again) doesn't make sense to me in general, but I may be overlooking something here.

Reply:
Thanks for pointing this out! I believe this is correct since we’re passing an array as defined by the `@psalm-type`

https://github.com/RRZE-Webteam/FAU-Studium-Common/blob/f9b1698f53be1d9882d6b9c9cdd4ccd7075cc2c2/src/Application/Link.php#L10-L17